### PR TITLE
blob contribution card implementation

### PR DESCRIPTION
**Pull request by jackyzha0**
- debounced update of shaderMaterial of blob on contrib change
- some abstraction of components
- computation of sha256 of wallet id, pattern, and response

### CODE REVIEW

File: browser/src/components/Blob.jsx [hunk 1]
```diff
@@ -52,7 +61,6 @@ export default function Blob({
     <mesh {...props}>
       <icosahedronGeometry attach="geometry" args={[size, 64, 64]} />
       <BlobShaderMaterial
-        size={size}
```
Comment (alejandrogarciasalas) [inline browser/src/components/Blob.jsx:55]: ah I think this was unused

File: browser/src/helpers/blobs.tsx [hunk 2]
```diff
@@ -1,9 +1,70 @@
...
+};
+
+export function renderBlob(contribution: Contribution): React.ReactElement {
+  // 64 chars long sha256 str
+  const contrib = `${contribution.author.walletId}@${contribution.pattern}: ${contribution.response}`;
+
+  const [message, setMessage] = useState(sha256(`0x${toHex(contrib)}`));
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setMessage(sha256(`0x${toHex(contrib)}`));
```
Comment (alejandrogarciasalas) [inline browser/src/helpers/blobs.tsx:36]: I guess this works, but I wonder if you should move the set state on the component above and rely on an onChange listener

File: browser/src/components/Blob.jsx [hunk 3]
```diff
@@ -33,10 +34,18 @@ export function BlobShaderMaterial({
       vertexShader,
     }),
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    [],
+    []
   );
 
-  return <shaderMaterial attach="material" {...data} />;
+  useFrame((state) => {
```
Comment (alejandrogarciasalas) [inline browser/src/components/Blob.jsx:40]: this will update the values every frame, why is it needed?

**Review by alejandrogarciasalas — APPROVED** [review]
approving to unblock lol, we can cleanup refactor later

File: browser/src/helpers/blobs.tsx [hunk 4]
```diff
@@ -1,9 +1,70 @@
...
+import {Placeholder} from "../components/ContributionSection";
+
+function toHex(str: string) {
+  return str
+    .split("")
+    .map((_, i) => str.charCodeAt(i).toString(16))
+    .join("");
+}
+
+function fetch(message: string, idx: number, val_start: number, range: number) {
```
Comment (alejandrogarciasalas) [inline browser/src/helpers/blobs.tsx:16]: why is this called `fetch` ? makes me think it's going to do an ajax call lol

File: browser/src/helpers/blobs.tsx [hunk 5]
```diff
@@ -1,9 +1,70 @@
...
+    }, 500);
+
+    return () => {
+      // teardown
+      clearTimeout(handler);
+    };
+  }, [contrib]);
+
+  return (
+    <Canvas camera={{ position: [0, 0, 20], fov: 50 }}>
```
Comment (alejandrogarciasalas) [inline browser/src/helpers/blobs.tsx:46]: I wonder how many threejs/webgl instances can we support rendering at a time w/o hitting perf issues, some other sites will just use a single canvas in the page and then move the camera on scroll/swap the objects but I think this is more straightforward

File: browser/src/helpers/blobs.tsx [hunk 6]
```diff
@@ -1,9 +1,70 @@
...
+  return float * range * eightCount + val_start;
+}
+
+export const PromptColours: Record<Prompt, number> = {
+    [Prompt.LooksLike]: 0,
+    [Prompt.WeNeed]: 0.25,
+    [Prompt.Example]: 0.5,
+};
+
+export function renderBlob(contribution: Contribution): React.ReactElement {
```
Comment (alejandrogarciasalas) [inline browser/src/helpers/blobs.tsx:29]: nit: rename this function and file

File: browser/src/helpers/blobs.tsx (hunk 2 above)
Comment (jackyzha0) [reply browser/src/helpers/blobs.tsx:36]: I think ideally this just debounces the display right? The internal value should be as up-to-date as possible

File: browser/src/components/Blob.jsx (hunk 3 above)
Comment (jackyzha0) [reply browser/src/components/Blob.jsx:40]: uniform doesn't update properly without it 🤷 I'm not super familiar with `react-three-fiber` but this is the thread I was going off of https://spectrum.chat/react-three-fiber/general/update-uniforms-of-shadermaterial-when-props-change~8dddcbd1-59e3-4a5a-996b-7c460c8e57ff

File: browser/src/helpers/blobs.tsx (hunk 4 above)
Comment (jackyzha0) [reply browser/src/helpers/blobs.tsx:16]: yeah my bad LOL was not thinking when I was naming

**Pull request merged**